File: medplum-java/medplum-model/src/main/java/com/fhir/MedicationAdministration_Performer.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationAdministration_Performer.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationAdministration_Performer {
  Optional<String> id();

  Optional<CodeableConcept> function();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Reference actor();

  static ImmutableMedicationAdministration_Performer.ActorBuildStage builder() {
    return ImmutableMedicationAdministration_Performer.builder();
  }
}

File: medplum-typescript/src/model/fhir/MedicationAdministration_Performer.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Reference } from "./Reference";

export interface MedicationAdministration_Performer {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    function: CodeableConcept | null | undefined;
    actor: Reference;
}
